test/jdk/java/security/Policy/Root/Root.java

```diff
@@ -22,27 +22,57 @@
  */
 
 /*
- *
+ * @test
  * @bug 4619757
  * @summary User Policy Setting is not recognized on Netscape 6
  *          when invoked as root.
- * @run main/manual Root
+ * @library /test/lib
+ * @run driver Root
  */
 
-/*
- * Place Root.policy in the root home directory (/),
- * as /.java.policy and run as test as root user.
- */
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
 
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
 import java.security.*;
 
 public class Root {
-    public static void main(String[] args) {
-        Policy p = Policy.getPolicy();
-        if (p.implies(Root.class.getProtectionDomain(), new AllPermission())) {
-            System.out.println("Test succeeded");
-        } else {
-            throw new SecurityException("Test failed");
+    private static final String SRC = System.getProperty("test.src");
+    private static final String ROOT = System.getProperty("user.home");
+    private static final Path SOURCE = Paths.get(SRC, "Root.policy");
+    private static final Path TARGET = Paths.get(ROOT, ".java.policy");
+    public static void main(String[] args) throws Exception {
+        Files.copy(SOURCE, TARGET, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Could you please use the testng framework for initial setup of test.

Reply:
This is a @driver test that copies the Root.policy file to the home directory before running the RootTest in a new JVM. The reason is that upon the start of the new JVM it loads the default system-wide policy file and the default user policy file which is the Root.policy we just copied. With the testng framework, there is no way to load the custom user policy file without reinstalling the security manager in the test, that said, it doesn't match what the manual test does.

Reply:
Discussed with Mahendra offline, testng framework `@BeforeTest` performs test setup in the client JVM before test starts in the other JVM. We can use testng framework in this case to handle the policy file operations.